behat/tests/bootstrap/v2/clientManagement/ClientManagementTrait.php

```diff
@@ -0,0 +1,343 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DigidepsBehat\v2\ClientManagement;
+
+use DateTime;
+use DigidepsBehat\v2\Common\UserDetails;
+
+trait ClientManagementTrait
+{
+    /**
+     * @When I search for an existing client by their first name
+     */
+    public function iSearchForExistingClientByFirstName()
+    {
+        $user = is_null($this->interactingWithUserDetails) ? $this->profAdminDeputyNotStartedDetails : $this->interactingWithUserDetails;
+        $this->searchForClientBy($user->getClientFirstName(), $user);
+    }
+
+    /**
+     * @When I search for an existing client by their last name
+     */
+    public function iSearchForExistingClientByLastName()
+    {
+        $user = is_null($this->interactingWithUserDetails) ? $this->profAdminDeputyNotStartedDetails : $this->interactingWithUserDetails;
+        $this->searchForClientBy($user->getClientLastName(), $user);
+    }
+
+    /**
+     * @When I search for an existing client by their case number
+     */
+    public function iSearchForExistingClientByCaseNumber()
+    {
+        $user = is_null($this->interactingWithUserDetails) ? $this->profAdminDeputyNotStartedDetails : $this->interactingWithUserDetails;
+        $this->searchForClientBy($user->getClientCaseNumber(), $user);
+    }
+
+    private function searchForClientBy(string $searchTerm, UserDetails $userDetailsInteractingWith)
+    {
+        $this->fillField('search_clients_q', $searchTerm);
+        $this->pressButton('Search');
+
+        $this->interactingWithUserDetails = $userDetailsInteractingWith;
+    }
+
+    /**
+     * @Then I should see the clients details in the client list results
+     */
+    public function iShouldSeeClientDetailsInResults()
+    {
+        $this->iShouldSeeNClientsWithSameName(1);
+    }
+
+    /**
+     * @Then I should see both the clients details in the client list results
+     */
+    public function iShouldSeeBothClientDetailsInResults()
+    {
+        $this->iShouldSeeNClientsWithSameName(2);
+    }
+
+    private function iShouldSeeNClientsWithSameName(int $numberClients)
```

Comment:
This is great and I don't know if it's going too far but should we also be checking for the 'Found N Clients' text. I know this has gone wrong before (template specifies number of clients and also does different things whether it plural or not. Eg Found 1 client VS Found 5 clilent**s**)

Reply:
Thats a good point actually - I'll add in a line to this scenario to check for it and another in the previous test for one client.